includes/admin/upgrades/migrations/migration-18_5.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * @package WPBDP\Admin\Upgrades\Migrations
+ */
+
+/**
+ * Migration for DB version 18.5
+ */
+class WPBDP__Migrations__18_5 extends WPBDP__Migration {
+
+	/**
+	 * If payments were on, we disable all free plans and vice versa.
+	 * We scan all enabled plans first for this migration.
+	 * If payments was enabled, we disable the free plans.
+	 * If payments was disabled, we disabled the paid plans.
+	 *
+	 * @since x.x
+	 */
+	public function migrate() {
+		global $wpdb;
+		$payments_on = wpbdp_get_option( 'payments-on' );
+		$sql         = "SELECT p.id, p.amount FROM {$wpdb->prefix}wpbdp_plans p WHERE p.enabled != 0";
+		$plans       = $wpdb->get_results( $sql );
+		$to_disable  = array();
+
+		if ( ! $plans ) {
+			return;
+		}
+
+		foreach ( $plans as $plan ) {
+			if ( $payments_on && $plan->amount <= 0.0 ) {
```

Comment:
This logic will change the plans that currently show on the page. When payments are on, the default free plan should be disabled. The other free plans should be left alone.

Please refer back to the migration details listed here:
https://github.com/Strategy11/business-directory-plugin/pull/65#issuecomment-959809227

Reply:
The current logic is :

- Payments on : all free plans are disabled
- Payments off : all paid plans disabled and free plans enabled

The code here takes care of that as we do not want a situation after migration to enable free plans on a site that had them disabled due to condition 1

Reply:
I'm still running tests on this logic with a fresh install with default settings and the updated settings applied to it

Reply:
But that's not the way it currently works. Can you please set up the case I mentioned for unit tests and see how it works in the current version? It's impossible not to break it if you're not super familiar with what should show.

Reply:
Yes, I am currently running various tests cases. This will be adjusted after I finish various tests with the main code and migrations triggered manually

Reply:
Working on the branch now for the additional tests